src/main/java/org/checkerframework/languageserver/CFTextDocumentService.java

```diff
@@ -154,13 +163,60 @@ public void didSave(DidSaveTextDocumentParams params) {
 
     @Override
     public void publish(Map<String, List<javax.tools.Diagnostic<?>>> result) {
+        typeRefinementMapping.clear();
         for (Map.Entry<String, List<javax.tools.Diagnostic<?>>> entry : result.entrySet()) {
             server.publishDiagnostics(
                     new PublishDiagnosticsParams(
                             entry.getKey(),
                             entry.getValue().stream()
+                                    .filter(x -> x.getKind() != javax.tools.Diagnostic.Kind.NOTE)
                                     .map(this::convertToLSPDiagnostic)
                                     .collect(Collectors.toList())));
+            entry.getValue().stream()
+                    .filter(x -> x.getKind() == javax.tools.Diagnostic.Kind.NOTE)
+                    .forEach(
+                            i -> {
+                                ComparablePosition start =
+                                        new ComparablePosition(
+                                                new Position(
+                                                        (int) i.getLineNumber() - 1,
+                                                        (int) i.getColumnNumber() - 1));
+                                String msg = i.getMessage(Locale.getDefault());
+                                int nameLength = msg.split("variable=")[1].split("\"")[1].length();
+                                ComparablePosition end =
+                                        new ComparablePosition(
+                                                new Position(
+                                                        (int) i.getLineNumber() - 1,
+                                                        ((int) i.getColumnNumber())
+                                                                + nameLength
+                                                                - 1));
+                                typeRefinementMapping.put(
+                                        com.google.common.collect.Range.closed(start, end), msg);
```

Comment:
I think the `variable="xxx", ` part can be discarded once hover retrieves the correct information.

Reply:
I have a couple of questions: 
1. How do I retrieve the current file name from the hover method and is the `entry.getKey()` the same value as the file name?
2. When displaying the information do we just want to show the values within the `types`  For example: 

`Sample1.java:30: Note: [type.refinements] variable="s", types=[@Initialized, @NonNull]
        var s = new Sample1();
            `

Would we only want to display `@Initialized, @NonNull`

Reply:
1. From the LSP specification, the hover request implements [TextDocumentPositionParams](https://microsoft.github.io/language-server-protocol/specifications/specification-3-17/#textDocumentPositionParams) which contains a `TextDocumentIdentifier`. I think the hover request object in lsp4j should also have a similar field. That being said, I'm not sure if the file names will share the same format, so please try it out. (Note: install [this java pluging](https://marketplace.visualstudio.com/items?itemName=redhat.java) and turn on `java.trace.server` in settings, and then you are able to see the logs from server.)
2. I have just updated the output format to `variable="s", checker="...", types=[...]`. I think we should display something like
```
checker="KeyFor", types=[@UnknownKeyFor]
checker="Nullness", types=[@Initialized, @NonNull]
```
so we have one line for each checker. In LSP specification, a Hover response can contain an array of `MarkedString`'s, so it should be pretty straightforward.

Reply:
> > Thank you for addressing my previous comments and making hover start to work! I think I've found a few bugs, so please check my comments below.
> > I also noticed that changing the default of `checker-framework.languageServerPath` in `package.json` doesn't work for me (it's still going to use the server jar under the download folder). Could you try this on your machine?
> 
> Hi, I believe it is also not working on my machine. I have been replacing the jar file within the download folder to test the hover.

Thanks for checking! I think we can fix this issue in a separate PR.